ipatests/test_integration/test_installation.py

```diff
@@ -847,3 +847,36 @@ def test_replicamanage_del(self):
 
         assert_text = 'Deleted IPA server "%s"' % self.replicas[0].hostname
         assert assert_text in cmd_output.stdout_text
+
+
+class TestInstallReplicaAgainstSpecificServer(IntegrationTest):
+    """Installation of replica against a specific server
+
+    Test to check replica install against specific server. It uses master and
+    replica1 without CA and having custodia service stopped. Then try to
+    install replica2 from replica1 so that replica2 will fetch secrets from
+    master as custodia service is not running on replica1.
+
+    related ticket: https://pagure.io/freeipa/issue/7566
+    """
+
+    num_replicas = 2
+
+    def test_replica_install_against_server(self):
+        tasks.install_master(self.master)
+
+        # install replica1 without CA
+        tasks.install_replica(self.master, self.replicas[0],
+                              setup_ca=False, setup_dns=True,
+                              promote=False)
+
+        # stop custodia service on replica1
+        self.replicas[0].run_command('systemctl stop ipa-custodia.service')
+
+        # check if custodia service is stopped
+        cmd = self.replicas[0].run_command('ipactl status')
+        assert 'ipa-custodia Service: STOPPED' in cmd.stdout_text
+
+        # install replica2 against replica1, as custodia service is stopped
+        # on replica1, replica2 will fetch secrets from master
+        tasks.install_replica(self.replicas[0], self.replicas[1])
```

Comment:
-  tasks.install_replica() defaults to promote=True which launches ipa-client-install first then ipa-replica-install. However promoting a client makes ipa-replica-install ignore the --server argument:
```
transport.py               558 DEBUG    IPA client is already configured on this system, ignoring the --domain, --server, --realm, --hostname, --password and --keytab options.
```

- The first master is installed with the default setup_dns=True but that second replica uses (again, the default) setup_dns=False so it is not a complete replica.

- With the custodia service stopped on replica0 I would _expect_ tasks.install_replica(self.replicas[0], self.replicas[1]) to fail.

Reply:
Replica1 is supposed to fetch the secrets from master in that case.